huaweicloud/services/dataarts/resource_huaweicloud_dataarts_studio_directory.go

```diff
@@ -0,0 +1,335 @@
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+
+	updateDirectoryOpt.JSONBody = utils.RemoveNil(buildUpdateDirectoryBodyParams(d))
+	updateDirectoryResp, err := updateDirectoryClient.Request("PUT", updateDirectoryPath, &updateDirectoryOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	updateDirectoryRespBody, err := utils.FlattenResponse(updateDirectoryResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	directory := utils.PathSearch("data.value", updateDirectoryRespBody, nil)
+
+	id, err := jmespath.Search("id", directory)
+	if err != nil {
+		return diag.Errorf("error updating DataArts Studio directory: %s is not found in API response", id)
+	}
+
+	qualifiedName, err := jmespath.Search("qualified_name", directory)
+	if err != nil {
+		return diag.Errorf("error updating DataArts Studio directory: %s is not found in API response", qualifiedName)
+	}
+	if qualifiedName == nil {
+		qualifiedName = d.Get("qualified_name")
+	}
+
+	rootID, err := jmespath.Search("root_id", directory)
+	if err != nil {
+		return diag.Errorf("error updating DataArts Studio directory: %s is not found in API response", rootID)
+	}
+
+	d.SetId(id.(string))
+	d.Set("root_id", rootID)
+	d.Set("qualified_name", qualifiedName)
+
+	return nil
+}
+func buildUpdateDirectoryBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"id":          d.Get("id"),
+		"name":        d.Get("name"),
+		"type":        d.Get("type"),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
+		"parent_id":   utils.ValueIngoreEmpty(d.Get("parent_id")),
+	}
+	return bodyParams
+}
+
+func resourceStudioDirectoryDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	//nolint:misspell
+	deleteDirectoryHttpUrl := "v2/{project_id}/design/directorys?ids={id}"
+	deleteDirectoryProduct := "dataartsv2"
+
+	deleteDirectoryClient, err := cfg.NewServiceClient(deleteDirectoryProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio V2 Client: %s", err)
+	}
+	deleteDirectoryPath := deleteDirectoryClient.Endpoint + deleteDirectoryHttpUrl
+	deleteDirectoryPath = strings.ReplaceAll(deleteDirectoryPath, "{project_id}", deleteDirectoryClient.ProjectID)
+	deleteDirectoryPath = strings.ReplaceAll(deleteDirectoryPath, "{id}", d.Get("id").(string))
+
+	deleteDirectoryOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		MoreHeaders:      map[string]string{"workspace": d.Get("workspace_id").(string)},
+	}
+
+	_, err = deleteDirectoryClient.Request("DELETE", deleteDirectoryPath, &deleteDirectoryOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceStudioDirectoryImportState(_ context.Context, d *schema.ResourceData, _ interface{}) (
+	[]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), "/", 3)
```

Comment:
we will use `Split` instead of `SplitN`

```suggestion
	parts := strings.Split(d.Id(), "/")
```